src/tools/pub/dependencies.md

```diff
@@ -397,23 +388,71 @@ dependency_overrides:
   transmogrify: '3.2.1'
 {% endprettify %}
 
-*Caution:* Using a dependency override involves some risk. For example,
-using an override to specify a version outside the range that the
-package claims to support, or using an override to specify
-a local copy of a package that has unexpected behaviors,
-may break your application.
+{{site.alert.warning}}
+  Using a dependency override involves some risk. For example,
+  using an override to specify a version outside the range that the
+  package claims to support, or using an override to specify
+  a local copy of a package that has unexpected behaviors,
+  may break your application.
+{{site.alert.end}}
+
+## Best practices
+
+It’s important to actively manage your dependencies and
+ensure that your packages use the freshest versions possible.
+If any dependency is stale,
+then you might have not only a stale version of that package,
+but also stale versions of other packages in your dependency graph that
+depend on that package.
+These stale versions can have a negative impact on
+the stability, performance, and quality of apps.
+
+We recommend the following best practices for package dependencies:
+
+Use [caret syntax](#caret-syntax)
+: Specifying dependencies with version ranges is such as `^1.6.3`
+  is a good practice because it allows the pub tool to
+  select newer versions of the package when they become available.
+  Also, it places an upper limit on the allowed version,
+  based on an assumption that packages use [semantic versions][],
+  where any version of path versioned `1.x` is compatible,
+  but where a new version `2.x` would be a major upgrade
+  that isn't semantically compatible with `1.x` versions. 
+
+Depend on the latest stable package versions
+: Use [`pub upgrade`][] to update to the latest package versions
+  that your pubspec allows.
+  To identify dependencies in your pubspec that
```

Comment:
This is not entirely accurate as `outdated` looks at both `pubspec.yaml` and `pubspec.lock`.

Maybe say something like "To identify dependencies in your app or package that are not on the latest possible versions, use pub outdated."